File: 15Feb2024.java
import java.util.Arrays;

class Solution {
    public long largestPerimeter(int[] nums) {
        Arrays.sort(nums);
        long ans = -1, sum = nums[0] + nums[1];
        for (int i = 2; i < nums.length; i++) {
            if (nums[i] < sum) {
                ans = Math.max(ans, sum + (long) nums[i]);
            }
            sum += (long) (nums[i]);
        }
        return ans;

    }
}

File: 15Feb2024.py
from ast import List


class Solution:
    def largestPerimeter(self, nums: List[int]) -> int:
        nums.sort()
        ans,sum=-1,nums[0]+nums[1]
        for i in range(2,len(nums)) :
            if(nums[i]<sum) :
                ans=max(ans,sum+nums[i])
            sum+=nums[i]
        
        return ans